ksp_plugin_adapter/utf8_marshaler.cs

```diff
@@ -6,38 +6,28 @@
 namespace principia {
 namespace ksp_plugin_adapter {
 
-internal abstract class UTF8Marshaler : ICustomMarshaler {
-  public abstract void CleanUpNativeData(IntPtr native_data);
-  public abstract IntPtr MarshalManagedToNative(object managed_object);
-  public abstract object MarshalNativeToManaged(IntPtr native_data);
-
-  void ICustomMarshaler.CleanUpManagedData(object managed_object) {}
-
-  int ICustomMarshaler.GetNativeDataSize() {
-    return -1;
-  }
-
+// A marshaler that knows how to encode/decode UTF-8 strings.
+internal abstract class UTF8Marshaler : MonoMarshaler {
   protected static readonly Encoding utf8_ =
       new UTF8Encoding(encoderShouldEmitUTF8Identifier : false,
                        throwOnInvalidBytes             : true);
 }
 
-// A marshaler for in parameter UTF-8 strings whose ownership is not taken from
-// the caller.
-internal class InUTF8Marshaler : UTF8Marshaler {
-  // In addition to implementing the |ICustomMarshaler| interface, custom
-  // marshalers must implement a static method called |GetInstance| that accepts
-  // a |String| as a parameter and has a return type of |ICustomMarshaler|,
-  // see https://goo.gl/wwmBTa.
+// A marshaler for UTF-8 strings whose ownership is not taken from C++.
```

Comment:
"Not taken from C++" is a property that only makes sense for results/out parameters: for in and in out parameters there is no C++ ownership to take.
Further, the marshaler name itself is misleading: for in and in out parameters, the string ownership is C#-side—and since the marshaler is doing the construction and destruction, it owns the string.

This really is a `NoTransferOfOwnershipUTF8Marshaler`: either the string may be owned by C++ or by C#, but it stays that way. That seems like too many words for an identifier; maybe `UncededUTF8Marshaler`?

Reply:
The verb "cede" in this context reads weird.  Let's discuss the name.

Reply:
Done.